unit_tests/test_aggregator.py

```diff
@@ -0,0 +1,415 @@
+# pylint: disable=missing-function-docstring
+from unittest.mock import patch, PropertyMock
+import pytest
+
+from gsy_e_sdk.aggregator import Aggregator, AggregatorWebsocketMessageReceiver
+from unit_tests.constants import (test_simulation,
+                                  test_agg, test_batch_command_dict,
+                                  test_response, TEST_AGGREGATOR_PREFIX,
+                                  TEST_CONFIGURATION_PREFIX)
+
+
+#####################################
+#            FIXTURES               #
+
+@pytest.fixture(autouse=True)
+def mock_connections(mocker):
```

Comment:
If I got it right, you're always mocking all these methods for all the tests that we're running in this module. The issue I see with this is that you're mocking the methods even for tests that don't need those mocks. Is this correct?
If this is the case, I would suggest to change the current structure. I think it would be better to only mock the necessary methods/objects for each test, and not to mock all of them at once. This will help with debugging, avoiding issues that might be hidden by mocks, and also to make it clearer which elements are involved in each individual test.

Please let me know if I got it right and if this makes sense to you.

Reply:
To go into more details. I see that one of the fixtures you used is named `mock_execute_batch_command_methods()`. So you are preparing all the mocks that are needed when testing the `execute_batch_command` methods. That's a good intuition in my opinion. However, the scope of the fixture is `function` by default, so these mocks will be applied to all tests, if I remember correctly, and not just to the ones that involve `execute_batch_command`. That's what I'm not convinced about.

Reply:
@ERafaelMartinez in this specific case, you can use `@pytest.mark.usefixtures('my_fixture')`, this way you'll have the fixture imported in your function without getting a pylint warning